### ISSUE TITLE
Components["List"] expects the ref to be an HTMLDivElement in Typescript [BUG]

### ISSUE THREAD

**Issue by allanRubenstein** [petyosi/react-virtuoso#864]
**Describe the bug**
When using typescript and a custom List, Typescript throws an error and says it expects an HTMLDivElement when you try to pass the ref to anything else, for example an unordered list.

**Reproduction**
https://codesandbox.io/s/virtualized-custom-list-and-item-type-issue-g3irvh?file=/App.tsx:415-433

**To Reproduce**
1. Create a component of type `Components["List"]` following the example on https://virtuoso.dev/customize-structure/
2. Try and pass the ref to an element other than a div
5. See error on the ref

**Expected behavior**
The ref is able to be passed to any html element.

**Screenshots**
![Screenshot 2023-03-16 at 4 11 52 PM](https://user-images.githubusercontent.com/33589855/225741683-7f8864cb-8d40-4139-af73-c2dbcdfb8b31.png)

**Comment by petyosi** [comment]
True, do you think you can send a PR for that? Worth doing so for the Items, too.

**Comment by singhAmandeep007** [comment]
@allanRubenstein 

Following fixed the types

```
type TVirtuosoListProps = {
  children: React.ReactNode;
} & React.HTMLAttributes<HTMLUListElement>

const VirtuosoList  = forwardRef<HTMLUListElement, TVirtuosoListProps>((props, ref) => {
  return (
    <ul
      aria-labelledby="locations-list-label"
      aria-describedby="locations-list-description"
      ref={ref}
      {...props}
    >
      {props.children}
    </ul>
  );
});
```

**Comment by singhAmandeep007** [comment]
@allanRubenstein 

Here is a work around 

```ts

const VirtuosoList: Components["List"] = forwardRef((props, ref) => {
  return (
    <ul
      aria-labelledby="locations-list-label"
      aria-describedby="locations-list-description"
      ref={ref as React.RefObject<HTMLUListElement>}
      {...(props as React.HTMLAttributes<HTMLUListElement>)}
    >
      {props.children}
    </ul>
  );
});

```

**Comment by aPinix** [comment]
> @allanRubenstein
> 
> Here is a work around
> 
> ```ts
> const VirtuosoList: Components["List"] = forwardRef((props, ref) => {
>   return (
>     <ul
>       aria-labelledby="locations-list-label"
>       aria-describedby="locations-list-description"
>       ref={ref as React.RefObject<HTMLUListElement>}
>       {...(props as React.HTMLAttributes<HTMLUListElement>)}
>     >
>       {props.children}
>     </ul>
>   );
> });
> ```

I can't seem to use this, at least in the version `"react-virtuoso": "^4.10.4"`.

I get an error **"Maximum update depth exceeded. This can happen when a component repeatedly calls setState inside componentWillUpdate or componentDidUpdate. React limits the number of nested updates to prevent infinite loops."**

Can anybody help? Thanks in advnce
